### PR TITLE
Update App.java

### PR DESCRIPTION
**Pull request by mbStavola**
Actually create Singletons now.

### CODE REVIEW

**Comment by mbStavola** [comment]
Prior to this change, you'd return a new component with each call. This meant that every time you provided a singleton, you were actually getting a new instance!

File: app/src/main/java/com/kevinmost/overlauncher/app/App.java [hunk 1]
```diff
@@ -7,9 +7,10 @@
 
 public class App extends Application {
   private static App INSTANCE;
-
+  private static AppComponent COMPONENT_INSTANCE = DaggerAppComponent.builder().appModule(INSTANCE.module).build();
```
Comment (kevinmost) [inline app/src/main/java/com/kevinmost/overlauncher/app/App.java]: Can you make this an instance variable instead? No need for member variables to be static considering we use a singleton instance of App anyway. And then your `provideComponent()` can return `INSTANCE.appComponentInstance`

Comment (mbStavola) [inline app/src/main/java/com/kevinmost/overlauncher/app/App.java]: Good recommendation.

File: app/src/main/java/com/kevinmost/overlauncher/app/App.java [hunk 2]
```diff
@@ -7,8 +7,9 @@
 
 public class App extends Application {
   private static App INSTANCE;
-
+  
   private final AppModule module = new AppModule(this);
+  private final AppComponent appComponentInstance = DaggerAppComponent.builder().appModule(INSTANCE.module).build();
```
Comment (kevinmost) [inline app/src/main/java/com/kevinmost/overlauncher/app/App.java]: Replace `INSTANCE.module` with just `module`. You're already within the same singleton instance, no need to specify that.

**Comment by kevinmost** [comment]
Nice, can you squash this down into one commit so I can merge it into master?

**Pull request closed without merging**